### PR TITLE
Fix spglib wrapper

### PR DESCRIPTION
**Pull request by antoine-levitt**
Fixes https://github.com/JuliaMolSim/DFTK.jl/issues/307

### CODE REVIEW

File: src/external/spglib.jl [hunk 1]
```diff
@@ -116,7 +113,7 @@ function spglib_standardize_cell(lattice::AbstractArray{T}, atoms; correct_symme
     # Ask spglib to standardize the cell (i.e. find a cell, which fits the spglib conventions)
     num_atoms = ccall((:spg_standardize_cell, SPGLIB), Cint,
       (Ptr{Cdouble}, Ptr{Cdouble}, Ptr{Cint}, Cint, Cint, Cint, Cdouble),
-      spg_lattice, spg_positions, spg_numbers, length(spg_numbers), Cint(primitive),
+      copy(spg_lattice'), spg_positions, spg_numbers, length(spg_numbers), Cint(primitive),
```
Comment (mfherbst) [inline src/external/spglib.jl:116]: This won't work, because spglib modifies the lattice in-place.

**Review by mfherbst — COMMENTED** [review]
Did you test this with the failing case?

**Comment by mfherbst** [comment]
Cool. Thanks for doing that. I'll add a test for the failing case in #307 before merging.

File: src/external/spglib.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/external/spglib.jl:116]: Good catch!

**Comment by antoine-levitt** [comment]
Great! (I hate tests)

**Comment by mfherbst** [comment]
Who doesn't, but I have the pipeline after all ...

**Comment by mfherbst** [comment]
Failing test on MacOS is the same issue as #304. I added a fix there ... let's hope that cuts it.

**Pull request merged**